sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -56,6 +57,11 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
             }
         }
 
+#if NET6_0_OR_GREATER
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(HttpMessage))]
```

Comment:
Do you really need `All`? That one is pretty broad and can easily increase the size of the app by a lot. For example, do you need to preserve interfaces and all members on the interfaces for this type?

Reply:
This is a hard one because this `RequestActivityPolicy` is built into our HttpPipeline - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L186
The HttpMessage is written using DiagnosticSource.Write() to signal start, stop, and failed events for all HTTP requests for all of the HTTP libraries in our repository. The HttpMessage is used as the value for these events. I'm not exactly sure which libraries in our repository make use of this and how. I will ask around about it and see if this can be narrowed.